src/components/core/common-behaviors/named-slot-state-controller.ts

```diff
@@ -20,15 +20,18 @@ import { ReactiveController, ReactiveControllerHost } from 'lit';
  * https://developer.mozilla.org/en-US/docs/Web/CSS/Attribute_selectors
  */
 export class NamedSlotStateController implements ReactiveController {
-  private _slots = new Set<string>();
+  public readonly slots = new Set<string>();
 
   // We avoid using AbortController here, as it would mean creating
   // a new instance for every NamedSlotStateController instance.
   private _slotchangeHandler = (event: Event): void => {
     this._syncSlots(event.target as HTMLSlotElement);
   };
 
-  public constructor(private _host: ReactiveControllerHost & Partial<HTMLElement>) {
+  public constructor(
+    private _host: ReactiveControllerHost & Partial<HTMLElement>,
+    private _onSlotStateChangeCallback = () => null,
```

Comment:
the changes here should not be needed. a callback is not necessary anymore. just use the var, it's updated automatically @TomMenga